Lib/http/cookiejar.py

```diff
@@ -1173,6 +1173,8 @@ def domain_return_ok(self, domain, request):
             req_host = "."+req_host
         if not erhn.startswith("."):
             erhn = "."+erhn
+        if not domain.startswith("."):
+            domain = "."+domain
```

Comment:
This will affect calls of `self.is_blocked(domain)` and `self.is_not_allowed(domain)` below.

Reply:
Thanks @serhiy-storchaka . My bad that I looked into fixing the issue and not about the underlying callers that use the dot-prefixed domain. Yes, adding the extra dot makes the comparison to fail where A has an extra dot at start due to my patch at https://github.com/python/cpython/blob/f30060dcd07cd53879226816512ea80bff0d0a78/Lib/http/cookiejar.py#L601 .

Sample program where the domain should be blocked

```python
import urllib
from http.cookiejar import DefaultCookiePolicy

policy = DefaultCookiePolicy(blocked_domains=['xxxfoo.co.jp'])
req = urllib.request.Request('https://xxxfoo.co.jp/')
print(policy.domain_return_ok('xxxfoo.co.jp', req))
```

```
➜  cpython git:(master) ✗ python3.7 /tmp/bar.py 
False
```

```
➜  cpython git:(bpo35121) ✗ ./python.exe /tmp/bar.py  
True
```

With patch this returns true but should be false since the domain is blocked and the prefix dot makes the comparison `.xxxfoo.co.jp == xxxfoo.co.jp` . One fix would be to use dot prefixed domain only for the checks at 

https://github.com/python/cpython/blob/f30060dcd07cd53879226816512ea80bff0d0a78/Lib/http/cookiejar.py#L1178

I think this needs to be fixed but I am also afraid I might accidentally break something here since the function itself received no changes since 2004.